src/main/java/org/example/jvspringbootfirstbook/service/category/CategoryServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package org.example.jvspringbootfirstbook.service.category;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.example.jvspringbootfirstbook.dto.category.CategoryDto;
+import org.example.jvspringbootfirstbook.dto.category.CreateCategoryRequestDto;
+import org.example.jvspringbootfirstbook.exception.EntityNotFoundException;
+import org.example.jvspringbootfirstbook.mapper.CategoryMapper;
+import org.example.jvspringbootfirstbook.model.Category;
+import org.example.jvspringbootfirstbook.repository.category.CategoryRepository;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll(pageable).stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        return categoryMapper.toDto(
+                categoryRepository.findById(id).orElseThrow(
+                        () -> new EntityNotFoundException("Category with id " + id + " not found")
+                )
+        );
+    }
+
+    @Override
+    public CategoryDto save(CreateCategoryRequestDto categoryDto) {
+        Category category = categoryMapper.toEntity(categoryDto);
+        return categoryMapper.toDto(categoryRepository.save(category));
+    }
+
+    @Override
+    public CategoryDto update(Long id, CreateCategoryRequestDto categoryDto) {
+        if (!categoryRepository.findById(id).isPresent()) {
+            throw new EntityNotFoundException("Category with id " + id + " not found");
+        }
```

Comment:
You are using `@PutMapping` annotation in your controller, so when id isn't found, it will just create a new one, are you aware of this action?